### DISCUSSION TITLE
startup error

### DISCUSSION THREAD

**Discussion by XTUxiongda** [minio/minio#15410] [Q&A] [answered]
I am trying to perform federation expansion, but the startup reports an error, as follows

Error: Disk: http://xx.xx.xx.xx:9000/data1 returned unformatted disk found (*fmt.wrapError)
       endpoint="http://xx.xx.xx.xx:9000/data1"
       3: internal/logger/logger.go:249:logger.LogAlwaysIf()
       2: cmd/prepare-storage.go:50:cmd.glob..func7.1()
       1: cmd/erasure-sets.go:223:cmd.(*erasureSets).connectDisks.func2()

Error: Write failed. Insufficient number of disks online (*errors.errorString)
      11: internal/logger/logger.go:259:logger.LogIf()
      10: cmd/erasure-encode.go:104:cmd.(*Erasure).Encode()
       9: cmd/erasure-object.go:1059:cmd.erasureObjects.putObject()
       8: cmd/erasure-object.go:875:cmd.erasureObjects.PutObject()
       7: cmd/erasure-sets.go:938:cmd.(*erasureSets).PutObject()
       6: cmd/config-common.go:79:cmd.saveConfig()
       5: cmd/erasure-server-pool-decom.go:485:cmd.poolMeta.save()
       4: cmd/erasure-server-pool-decom.go:569:cmd.(*erasureServerPools).Init()
       3: cmd/erasure-server-pool.go:144:cmd.newErasureServerPools()
       2: cmd/server-main.go:694:cmd.newObjectLayer()
       1: cmd/server-main.go:531:cmd.serverMain()

Error: Unable to initialize backend: Storage resources are insufficient for the write operation .minio.sys/tmp/a8458874-8a8c-400e-b575-0082f8319f51/c29e0589-6d79-4b43-8ca8-cacd8dc5e80a/part.1, retrying in 1.943093444s (*fmt.wrapError)
       4: internal/logger/logger.go:259:logger.LogIf()
       3: cmd/erasure-server-pool.go:150:cmd.newErasureServerPools()
       2: cmd/server-main.go:694:cmd.newObjectLayer()
       1: cmd/server-main.go:531:cmd.serverMain()

**Comment by klauspost** [comment]
Follow https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html

Or provide actually useful information on your setup, what you've done, etc.

**Reply by XTUxiongda** [reply]
`#!/bin/bash
export MINIO_ROOT_USER=minioadmin
export MINIO_ROOT_PASSWORD=minioadmin

/opt/minio/minio server --console-address ":9001" --address ":9000"> minio.log \
http://10.105.100.61/data1 http://10.105.100.61/data2 \
http://10.105.100.62/data1 http://10.105.100.62/data2 \
http://10.105.100.63/data1 http://10.105.100.63/data2`

i want to deploy minio in distrubuted mode, i have 3 node (61,62,63), each server have 2 dirves .
when i start up , i meet the problem as follows:

API: SYSTEM()
Time: 08:43:20 UTC 07/27/2022
Error: Write failed. Insufficient number of disks online (*errors.errorString)
      11: internal/logger/logger.go:259:logger.LogIf()
      10: cmd/erasure-encode.go:104:cmd.(*Erasure).Encode()
       9: cmd/erasure-object.go:1059:cmd.erasureObjects.putObject()
       8: cmd/erasure-object.go:875:cmd.erasureObjects.PutObject()
       7: cmd/erasure-sets.go:938:cmd.(*erasureSets).PutObject()
       6: cmd/config-common.go:79:cmd.saveConfig()
       5: cmd/erasure-server-pool-decom.go:485:cmd.poolMeta.save()
       4: cmd/erasure-server-pool-decom.go:569:cmd.(*erasureServerPools).Init()
       3: cmd/erasure-server-pool.go:144:cmd.newErasureServerPools()
       2: cmd/server-main.go:694:cmd.newObjectLayer()
       1: cmd/server-main.go:531:cmd.serverMain()

API: SYSTEM()
Time: 08:43:20 UTC 07/27/2022
Error: Unable to initialize backend: Storage resources are insufficient for the write operation .minio.sys/tmp/f102f520-8b41-43d1-afd2-2cc8b213bba3/0f1d3fdb-3264-42ee-aee4-b195c292af49/part.1, retrying in 1.820252881s (*fmt.wrapError)
       4: internal/logger/logger.go:259:logger.LogIf()
       3: cmd/erasure-server-pool.go:150:cmd.newErasureServerPools()
       2: cmd/server-main.go:694:cmd.newObjectLayer()
       1: cmd/server-main.go:531:cmd.serverMain()

API: SYSTEM()
Time: 08:43:21 UTC 07/27/2022
Error: Disk: http://10.105.100.62:9000/data1 returned unformatted disk found (*fmt.wrapError)
       endpoint="http://10.105.100.62:9000/data1"
       3: internal/logger/logger.go:249:logger.LogAlwaysIf()
       2: cmd/prepare-storage.go:50:cmd.glob..func7.1()
       1: cmd/erasure-sets.go:223:cmd.(*erasureSets).connectDisks.func2()

API: SYSTEM()
Time: 08:43:21 UTC 07/27/2022
Error: Disk: http://10.105.100.63:9000/data1 returned unformatted disk found (*fmt.wrapError)
       endpoint="http://10.105.100.63:9000/data1"
       3: internal/logger/logger.go:249:logger.LogAlwaysIf()
       2: cmd/prepare-storage.go:50:cmd.glob..func7.1()
       1: cmd/erasure-sets.go:223:cmd.(*erasureSets).connectDisks.func2()

API: SYSTEM()
Time: 08:43:22 UTC 07/27/2022
Error: Write failed. Insufficient number of disks online (*errors.errorString)
      11: internal/logger/logger.go:259:logger.LogIf()
      10: cmd/erasure-encode.go:104:cmd.(*Erasure).Encode()
       9: cmd/erasure-object.go:1059:cmd.erasureObjects.putObject()
       8: cmd/erasure-object.go:875:cmd.erasureObjects.PutObject()
       7: cmd/erasure-sets.go:938:cmd.(*erasureSets).PutObject()
       6: cmd/config-common.go:79:cmd.saveConfig()
       5: cmd/erasure-server-pool-decom.go:485:cmd.poolMeta.save()
       4: cmd/erasure-server-pool-decom.go:569:cmd.(*erasureServerPools).Init()
       3: cmd/erasure-server-pool.go:144:cmd.newErasureServerPools()
       2: cmd/server-main.go:694:cmd.newObjectLayer()
       1: cmd/server-main.go:531:cmd.serverMain()

API: SYSTEM()
Time: 08:43:22 UTC 07/27/2022
Error: Unable to initialize backend: Storage resources are insufficient for the write operation .minio.sys/tmp/2ae54594-a74f-4e69-9988-e098f0dbda35/f0ddd462-74a2-41fc-afd8-0ff5627bf7e7/part.1, retrying in 4.103309814s (*fmt.wrapError)
       4: internal/logger/logger.go:259:logger.LogIf()
       3: cmd/erasure-server-pool.go:150:cmd.newErasureServerPools()
       2: cmd/server-main.go:694:cmd.newObjectLayer()
       1: cmd/server-main.go:531:cmd.serverMain()

API: SYSTEM()
Time: 08:43:26 UTC 07/27/2022
Error: Write failed. Insufficient number of disks online (*errors.errorString)
      11: internal/logger/logger.go:259:logger.LogIf()
      10: cmd/erasure-encode.go:104:cmd.(*Erasure).Encode()
       9: cmd/erasure-object.go:1059:cmd.erasureObjects.putObject()
       8: cmd/erasure-object.go:875:cmd.erasureObjects.PutObject()
       7: cmd/erasure-sets.go:938:cmd.(*erasureSets).PutObject()
       6: cmd/config-common.go:79:cmd.saveConfig()
       5: cmd/erasure-server-pool-decom.go:485:cmd.poolMeta.save()
       4: cmd/erasure-server-pool-decom.go:569:cmd.(*erasureServerPools).Init()
       3: cmd/erasure-server-pool.go:144:cmd.newErasureServerPools()
       2: cmd/server-main.go:694:cmd.newObjectLayer()
       1: cmd/server-main.go:531:cmd.serverMain()

**Reply by ravindk89** [reply]
```
/opt/minio/minio server --console-address ":9001" --address ":9000"> minio.log
http://10.105.100.61/data1 http://10.105.100.61/data2
http://10.105.100.62/data1 http://10.105.100.62/data2
http://10.105.100.63/data1 [http://10.105.100.63/data2`](http://10.105.100.63/data2%60)
```

What was the original command you were running here, pre-expansion?

We haven't really supported this style of deployment or expansion in a long time. How old is this setup? What version of MinIO are you running?

**Reply by harshavardhana** [reply]
@ravindk89 this won't work really - it is incorrectly setup.

**Reply by ravindk89** [reply]
That's my assumption as well - if this is a fresh setup @XTUxiongda there is no reason to use this kind of syntax. What was the intent here?

**Reply by XTUxiongda** [reply]
> ```
> /opt/minio/minio server --console-address ":9001" --address ":9000"> minio.log
> http://10.105.100.61/data1 http://10.105.100.61/data2
> http://10.105.100.62/data1 http://10.105.100.62/data2
> http://10.105.100.63/data1 [http://10.105.100.63/data2`](http://10.105.100.63/data2%60)
> ```
> 
> What was the original command you were running here, pre-expansion?
> 
> We haven't really supported this style of deployment or expansion in a long time. How old is this setup? What version of MinIO are you running?

```
#!/bin/bash
export MINIO_ROOT_USER=minioadmin
export MINIO_ROOT_PASSWORD=minioadmin
export MINIO_ETCD_ENDPOINTS="http://10.105.100.61:2379"
export MINIO_PUBLIC_IPS=10.106.0.56,10.106.0.57
export MINIO_DOMAIN=fofund.test.com

/opt/minio/minio server --console-address ":9001" --address ":9000"> minio.log \
http://10.106.0.57/data/data1 http://10.106.0.57/data2 \
http://10.106.0.56/data/data1 http://10.106.0.56/data2
```
this is the original command. and i find it is ok. then, i want to expand use etcd and coredns，so i add 3 nodes and script as above

**Reply by XTUxiongda** [reply]
i use the lastest release version, download from https://dl.min.io/server/minio/release/linux-amd64/minio

**Reply by XTUxiongda** [reply]
and i also want to ask a question, is this kind syntax not support or i can not use etcd to expand my minio cluster？

**Comment by ravindk89** [comment, marked as answer]
@XTUxiongda https://docs.min.io/minio/baremetal/installation/deploy-minio-distributed.html <- this document describes how to deploy a Distributed MinIO Deployment. There are also docs for expansion.

Using ETCD and the federated style approach was deprecated a long while back. Please reference our current docs moving forward.
